src/features/txnDelegatation/useHeliumTxns.ts

```diff
@@ -0,0 +1,136 @@
+import { AddGateway, Location, Transfer } from '@helium/react-native-sdk'
+import { SignHotspotResponse } from '@helium/wallet-link'
+import { useCallback, useMemo } from 'react'
+import { useSubmitTxnMutation } from '../../generated/graphql'
+import { getKeypair } from '../../storage/secureStorage'
+
+const useHeliumTxns = (
+  heliumAddress: string,
+  opts: {
+    addGatewayTxn?: string
+    assertLocationTxn?: string
+    transferHotspotTxn?: string
+  },
+) => {
+  const [submitTxnMutation, { loading: submitLoading }] = useSubmitTxnMutation()
+
+  const addGatewayTxn = useMemo(() => {
+    if (!opts.addGatewayTxn) return
+
+    return AddGateway.txnFromString(opts.addGatewayTxn)
+  }, [opts.addGatewayTxn])
+
+  const assertLocationTxn = useMemo(() => {
+    if (!opts.assertLocationTxn) return
+    return Location.txnFromString(opts.assertLocationTxn)
+  }, [opts.assertLocationTxn])
+
+  const transferHotspotTxn = useMemo(() => {
+    if (!opts.transferHotspotTxn) return
+
+    return Transfer.txnFromString(opts.transferHotspotTxn)
+  }, [opts.transferHotspotTxn])
+
+  const gatewayAddress = useMemo(() => {
+    return (
+      addGatewayTxn?.gateway?.b58 ||
+      assertLocationTxn?.gateway?.b58 ||
+      transferHotspotTxn?.gateway?.b58
+    )
+  }, [addGatewayTxn, assertLocationTxn, transferHotspotTxn])
+
+  const hasTransactions = useMemo(() => !!gatewayAddress, [gatewayAddress])
+
+  const submit = useCallback(() => {
+    // TODO: Confirm this works
+    const txnObject = assertLocationTxn || transferHotspotTxn
+    if (!txnObject) return
+    return submitTxnMutation({
+      variables: {
+        address: heliumAddress,
+        txn: txnObject?.toString(),
+        txnJson: JSON.stringify(txnObject),
```

Comment:
Maybe just send the `txn` as a string and the maker can stringify on the other end.

```suggestion
      variables: {
        address: heliumAddress,
        txn: txnObject?.toString()
```

Reply:
This is a mutation that's specific to our v1 api so that we can show pending txns in the app.This whole hook will get removed after the transition.